deeppavlov/models/nemo/asr.py

```diff
@@ -0,0 +1,193 @@
+# Copyright 2020 Neural Networks and Deep Learning lab, MIPT
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from io import BytesIO
+from pathlib import Path
+from typing import List, Optional, Tuple, Union, Dict
+
+import torch
+from nemo.collections.asr import AudioToMelSpectrogramPreprocessor, JasperEncoder, JasperDecoderForCTC, GreedyCTCDecoder
+from nemo.collections.asr.helpers import post_process_predictions
+from nemo.collections.asr.parts.features import WaveformFeaturizer
+from nemo.core.neural_types import AudioSignal, NeuralType, LengthsType
+from nemo.utils.decorators import add_port_docs
+from torch import Tensor
+from torch.utils.data import Dataset, DataLoader
+
+from deeppavlov.core.common.registry import register
+from deeppavlov.models.nemo.common import CustomDataLayerBase, NeMoBase
+
+log = logging.getLogger(__name__)
+
+
+class AudioInferDataset(Dataset):
+    def __init__(self, audio_batch: List[Union[str, BytesIO]], sample_rate: int, int_values: bool, trim=False) -> None:
+        """Dataset reader for AudioInferDataLayer.
+
+        Args:
+            audio_batch: Batch to be red. Elements could be either paths to audio files or Binary I/O objects.
+            sample_rate: Audio files sample rate.
+            int_values: If true, load samples as 32-bit integers.
+            trim: Trim leading and trailing silence from an audio signal if True.
```

Comment:
trim_silence probably?

Reply:
I decided to use the same argument name that uses a similar class to maintain uniformity: ``nemo.collections.asr.parts.dataset.AudioDataset`` uses ``trim`` argument name, as well as ``nemo.collections.asr.parts.features.WaveformFeaturizer.process`` method.